integration/query_test.go

```diff
@@ -613,3 +614,211 @@ func TestQueryNonExistingCollection(t *testing.T) {
+		skip       string              // optional, skip test with a specified reason
+	}{
+		"Int": {
+			batchSize:  1,
+			firstBatch: bson.A{bson.D{{"_id", int32(0)}}},
+			skip:       "https://github.com/FerretDB/FerretDB/issues/2005",
+		},
+		"Long": {
+			batchSize: int64(2),
+			firstBatch: bson.A{
+				bson.D{{"_id", int32(0)}},
+				bson.D{{"_id", int32(1)}},
+			},
+			skip: "https://github.com/FerretDB/FerretDB/issues/2005",
+		},
+		"LongZero": {
+			batchSize:  int64(0),
+			firstBatch: bson.A{},
+		},
+		"LongNegative": {
+			batchSize: int64(-1),
+			err: &mongo.CommandError{
+				Code:    51024,
+				Name:    "Location51024",
+				Message: "BSON field 'batchSize' value must be >= 0, actual value '-1'",
+			},
+		},
+		"DoubleNegative": {
+			batchSize: -1.1,
+			err: &mongo.CommandError{
+				Code:    51024,
+				Name:    "Location51024",
+				Message: "BSON field 'batchSize' value must be >= 0, actual value '-1'",
+			},
+		},
+		"DoubleFloor": {
+			batchSize:  1.9,
+			firstBatch: bson.A{bson.D{{"_id", int32(0)}}},
+			skip:       "https://github.com/FerretDB/FerretDB/issues/2005",
+		},
+		"Bool": {
+			batchSize:  true,
+			firstBatch: bson.A{bson.D{{"_id", int32(0)}}},
+			err: &mongo.CommandError{
+				Code:    14,
+				Name:    "TypeMismatch",
+				Message: "BSON field 'FindCommandRequest.batchSize' is the wrong type 'bool', expected types '[long, int, decimal, double']",
+			},
+			skip: "https://github.com/FerretDB/FerretDB/issues/2005",
+		},
+		"Unset": {
+			// default batchSize is 101 when unset
+			batchSize:  nil,
+			firstBatch: generateDocuments(0, 101),
+			skip:       "https://github.com/FerretDB/FerretDB/issues/2005",
+		},
+		"LargeBatchSize": {
+			batchSize:  102,
+			firstBatch: generateDocuments(0, 102),
+			skip:       "https://github.com/FerretDB/FerretDB/issues/2005",
+		},
+	} {
+		name, tc := name, tc
+		t.Run(name, func(t *testing.T) {
+			if tc.skip != "" {
+				t.Skip(tc.skip)
+			}
+
+			t.Parallel()
+
+			var rest bson.D
+			if tc.batchSize != nil {
+				rest = append(rest, bson.E{Key: "batchSize", Value: tc.batchSize})
+			}
+
+			command := append(
+				bson.D{{"find", collection.Name()}},
+				rest...,
+			)
+
+			var res bson.D
+			err := collection.Database().RunCommand(ctx, command).Decode(&res)
+			if tc.err != nil {
+				assert.Nil(t, res)
+				AssertEqualAltCommandError(t, *tc.err, tc.altMessage, err)
+
+				return
+			}
+
+			require.NoError(t, err)
+
+			v, ok := res.Map()["cursor"]
+			require.True(t, ok)
+
+			cursor, ok := v.(bson.D)
+			require.True(t, ok)
+
+			// Do not check the value of cursor id, FerretDB has a different id.
+			cursorID := cursor.Map()["id"]
+			assert.NotNil(t, cursorID)
+
+			firstBatch, ok := cursor.Map()["firstBatch"]
+			require.True(t, ok)
+			require.Equal(t, tc.firstBatch, firstBatch)
+		})
+	}
+}
+
+func TestQueryBatchSize(t *testing.T) {
+	t.Skip("https://github.com/FerretDB/FerretDB/issues/2005")
```

Comment:
Tests are currently skipped, but if we prefer I can make it not skipped on MongoDB only